src/Storages/System/StorageSystemDroppedTablesParts.cpp

```diff
@@ -0,0 +1,89 @@
+#include <Storages/StorageMaterializedMySQL.h>
+#include <Storages/VirtualColumnUtils.h>
+#include <Access/ContextAccess.h>
+#include <Storages/System/StorageSystemDroppedTablesParts.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeUUID.h>
+
+
+namespace DB
+{
+
+
+StoragesDroppedInfoStream::StoragesDroppedInfoStream(const SelectQueryInfo & query_info, ContextPtr context)
+        : StoragesInfoStreamBase(context)
+{
+    needsLock = false;
```

Comment:
It looks ugly, however, acquiring the lock for share (especially in Atomic database, especially if the table is dropped) is almost no-op (no chance it will wait for something)

Reply:
this class is common for both - dropped and alive tables - we need the way to distinct between just-dropped table for system.parts and table for system.dropped_tables_parts
but yes, it's ugly - will try to improve